smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/HttpQueryTraitValidator.java

```diff
@@ -77,8 +85,39 @@ private List<ValidationEvent> validateBindings(Map<StructureShape, Map<String, S
                             paramsToMembers.getKey(), ValidationUtils.tickedList(paramsToMembers.getValue()))));
                 }
             }
+
+            List<OperationShape> operations = structureToOperations.getOrDefault(entry.getKey(),
+                                                                                 Collections.emptyList());
+            for (OperationShape operation : operations) {
+                UriPattern pattern = operation.expectTrait(HttpTrait.class).getUri();
+                for (Map.Entry<String, String> literalEntry : pattern.getQueryLiterals().entrySet()) {
+                    String literalKey = literalEntry.getKey();
+                    if (entry.getValue().containsKey(literalKey)) {
+                        events.add(error(entry.getKey(), String.format(
+                            "`httpQuery` parameter name binding conflicts found for the `%s` parameter, the "
+                            + "http trait for the `%s` operation defines a query literal with the same name",
+                            literalKey, operation.getId())));
+                    }
+                }
+            }
         }
 
         return events;
     }
+
+    private Map<StructureShape, List<OperationShape>> getStructureToOperations(Model model) {
+        Map<StructureShape, List<OperationShape>> structureToOperations = new HashMap<>();
+        for (ServiceShape service : model.getServiceShapes()) {
+            for (OperationShape operation : TopDownIndex.of(model).getContainedOperations(service)) {
+                if (operation.hasTrait(HttpTrait.class)) {
+                    model.expectShape(operation.getInputShape())
```

Comment:
Have you considered using an `OperationIndex` here? 
I think that would look like: 
```
operationIndex.getInput(operation).ifPresent(inputStructure -> ...)
```

Reply:
Sure, I made the change accordingly.